docs/docs/how-tos/memory/shared-state.ipynb

```diff
@@ -0,0 +1,439 @@
+    "model = ChatOpenAI(model=\"gpt-4o-mini\").bind_tools([Info])\n",
+    "\n",
+    "\n",
+    "def call_model(state: MessagesState, config: RunnableConfig, *, store: BaseStore):\n",
+    "    \"\"\"Call the model.\"\"\"\n",
+    "    user_id = config[\"configurable\"][\"user_id\"]\n",
+    "    memories = store.search((\"memories\", user_id))\n",
+    "    info = \"\\n\".join([d.value[\"fact\"] for d in memories])\n",
+    "    # Format system prompt\n",
+    "    system_msg = prompt.format(info=info)\n",
+    "    # Call model\n",
+    "    response = model.invoke(\n",
+    "        [{\"role\": \"system\", \"content\": system_msg}] + state[\"messages\"]\n",
+    "    )\n",
+    "    return {\"messages\": [response]}\n",
+    "\n",
+    "\n",
+    "# Routing function to decide what to do next\n",
+    "# If no tool calls, then we end\n",
+    "# If tool calls, then we update memory\n",
+    "def route(state):\n",
+    "    if len(state[\"messages\"][-1].tool_calls) == 0:\n",
+    "        return END\n",
+    "    else:\n",
+    "        return \"update_memory\"\n",
+    "\n",
+    "\n",
+    "def update_memory(state: MessagesState, config: RunnableConfig, *, store: BaseStore):\n",
+    "    \"\"\"Update the memory.\"\"\"\n",
+    "    user_id = config[\"configurable\"][\"user_id\"]\n",
+    "    memory_id = str(uuid.uuid4())\n",
+    "    tool_calls = []\n",
+    "    memories = {}\n",
+    "    # Each tool call is a new memory to save\n",
+    "    for tc in state[\"messages\"][-1].tool_calls:\n",
+    "        # We append ToolMessages (to pass back to the LLM)\n",
+    "        # This is needed because OpenAI requires each tool call be followed by a ToolMessage\n",
+    "        tool_calls.append(\n",
+    "            {\"role\": \"tool\", \"content\": \"Saved!\", \"tool_call_id\": tc[\"id\"]}\n",
+    "        )\n",
+    "        # We create a new memory from this tool call\n",
+    "        store.put((\"memories\", user_id), memory_id, {\n",
+    "            \"fact\": tc[\"args\"][\"fact\"],\n",
+    "            \"topic\": tc[\"args\"][\"topic\"],\n",
+    "        })\n",
+    "    # Return the messages and memories to update the state with\n",
+    "    return {\"messages\": tool_calls}\n",
+    "\n",
+    "\n",
+    "# This is the in memory checkpointer we will use\n",
+    "# We need this because we want to enable threads (conversations)\n",
+    "checkpointer = MemorySaver()\n",
+    "\n",
+    "# This is the in memory store needed to save the memories (i.e. user preferences)\n",
+    "in_memory_store = InMemoryStore()\n",
+    "\n",
+    "# Construct this relatively simple graph\n",
+    "graph = StateGraph(MessagesState)\n",
+    "graph.add_node(call_model)\n",
+    "graph.add_node(update_memory)\n",
+    "graph.add_edge(\"update_memory\", END)\n",
+    "graph.add_edge(START, \"call_model\")\n",
+    "graph.add_conditional_edges(\"call_model\", route, [END, \"update_memory\"])\n",
```

Comment:
i dont think you need the third argument (can just type the conditional edge better)

if you do want to keep here, at least pass it as a kwarg (since it is)

Reply:
we're actually moving away from types in conditional edges to use this format instead in the guides, since you can't use `END` in `Literal` for type annotations, i.e. have to use `"__end__"`. so this is just part of the bigger change to standardize everything and avoid users having to type `__end__` everywhere

but lmk if you disagree with this!